chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -1920,8 +1960,13 @@ impl PeerManagerActor {
     #[perf]
     fn handle_msg_inbound_tcp_connect(&self, msg: InboundTcpConnect, ctx: &mut Context<Self>) {
         let _d = delay_detector::DelayDetector::new(|| "inbound tcp connect".into());
-
-        if self.is_inbound_allowed() {
+        if self.is_inbound_allowed()
+            || msg
+                .stream
+                .peer_addr()
+                .map(|addr| self.is_ip_whitelisted(&addr.ip()))
```

Comment:
Why aren’t we checking port? When we check is_peer_whitelisted we check full address, i.e. ip and port. We need to either check port in both or in neither. Currently we may end up accepting connection here but then dropping it later just because port didn’t match.

Reply:
These are different ports:
When the TCP connection is established, we get the TCP port of the outbound connection of the peer (not predictable)
During the handshake we obtain the peer identity, which includes the port on which the peer is listening for incoming connections (predictable).